File: Data Structures & Algorithms/meeting-schedule-ii/submission-1.java
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

class Solution {
    public int minMeetingRooms(List<Interval> intervals) {
        if (intervals == null) {
            return 0;
        }
        int N = intervals.size();
        int[] starts = new int[N];
        int[] ends = new int[N];

        for (int i = 0 ; i < N ; i++) {
            starts[i] = intervals.get(i).start;
            ends[i] = intervals.get(i).end;
        }
        Arrays.sort(starts);
        Arrays.sort(ends);

        int s = 0, e = 0, count = 0, res = 0;
        while (s < N && e < N) {
            if (starts[s] < ends[e]) {
                count++;
                s++;
            } else {
                e++;
                count--;
            }
            res = Math.max(count, res);
        }

        return res;
    }
}

File: Data Structures & Algorithms/merge-intervals/submission-1.ts
class Solution {
    /**
     * @param {number[][]} intervals
     * @return {number[][]}
     */
    merge(intervals: number[][]): number[][] {

        intervals.sort((a, b) => {
            if (a[0] == b[0]) return a[1] - b[1]
            return a[0] - b[0]
        })

        let res: number[][] = []
        let l = 0, r = 1

        res.push([...intervals[l]])
        while (r < intervals.length) {
            if (intervals[r][0] > res[res.length-1][1]) {
                res.push(intervals[r])
            } else {
                res[res.length-1][1] = Math.max(intervals[r][1], res[res.length-1][1])
            }
            r++
        }

        return res
    }
}
